### PR TITLE
Bug 1235128 - [TV][2.5] System banner supports icon and title

### PR DESCRIPTION
**Pull request by rickychien**
https://bugzilla.mozilla.org/show_bug.cgi?id=1235128

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6cff7164f452e6d101e9396d54a47b25411b2f96

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=963e684c1f03a948b0c05b31eedf937865221791

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d4bc571a97a03a9f1e5a1f85f693ddf85ca83b47

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f1ea4f6fe091d1392891cd8afdd7ee87154a8499

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=36f508b0400daa0c5f835fc332e2f1a20c7f5cae

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e7b6cd2f6058f95490352ee5f6212d9f910a170a

File: tv_apps/smart-system/js/app_install_manager.js [hunk 1]
```diff
@@ -409,15 +409,21 @@ var AppInstallManager = {
     var manifest = app.manifest || app.updateManifest;
     var appManifest = new ManifestHelper(manifest);
     var name = appManifest.name;
-    var msgID = this.isMarketplaceAppActive() ?
-      'added-to-apps' : 'app-install-success';
+    var msg = this.isMarketplaceAppActive() ? {
```
Comment (luke-chang) [inline tv_apps/smart-system/js/app_install_manager.js]: I suggest using regular `if ... else`. That would be clearer.

File: tv_apps/smart-system/js/app_install_manager.js [hunk 2]
```diff
@@ -517,11 +523,14 @@ var AppInstallManager = {
...
-          args: { appName: name }
-        };
-        this.systemBanner.show(msg);
+
+        this.systemBanner.show({
+          title: name,
+          text: {
+            id: 'app-install-' + key
+          },
+          icon: '/style/icons/add_to_apps.png'
```
Comment (luke-chang) [inline tv_apps/smart-system/js/app_install_manager.js]: I would suggest not using icon here.

File: tv_apps/smart-system/js/browser_context_menu.js [hunk 3]
```diff
@@ -330,11 +331,12 @@
...
-                id: 'deleted-from-apps',
-                args: {
-                  appName: name
-                }
+              this.systemBanner.show({
+                title: name,
+                text: {
+                  id: 'deleted-from-apps'
+                },
+                icon: '/style/icons/delete_from_apps.png'
```
Comment (luke-chang) [inline tv_apps/smart-system/js/browser_context_menu.js]: You can refer to `DELETE_FROM_APPS_ICON_PATH` constant.

File: tv_apps/smart-system/js/browser_context_menu.js [hunk 4]
```diff
@@ -360,11 +362,12 @@
...
-                  id: 'added-to-apps',
-                  args: {
-                    appName: name
-                  }
+                this.systemBanner.show({
+                  title: name,
+                  text: {
+                    id: 'added-to-apps'
+                  },
+                  icon: '/style/icons/add_to_apps.png'
```
Comment (luke-chang) [inline tv_apps/smart-system/js/browser_context_menu.js]: Again, use the constant here.

File: tv_apps/smart-system/js/browser_context_menu.js [hunk 5]
```diff
@@ -393,11 +396,12 @@
...
-                    id: 'deleted-from-apps',
-                    args: {
-                      appName: bookmark.name
-                    }
+                  this.systemBanner.show({
+                    title: bookmark.name,
+                    text: {
+                      id: 'delete-from-apps'
+                    },
+                    icon: '/style/icons/delete_from_apps.png'
```
Comment (luke-chang) [inline tv_apps/smart-system/js/browser_context_menu.js]: Again, use the constant here.

File: tv_apps/smart-system/bower_components/smart-banner/script.js [hunk 6]
```diff
@@ -119,9 +127,20 @@ window.SmartBanner = (function(win) {
...
   };
 
-  proto.flyOpen = function() {
+  proto.flying = function() {
+    this.classList.add('flying');
+  };
+
+  proto.bouncing = function() {
+    this.classList.remove('closed');
+    this.style.width = 'auto';
```
Comment (luke-chang) [inline tv_apps/smart-system/bower_components/smart-banner/script.js]: Should we reset `this.style.width` when it's closed?

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5a9da0ff86a8ed20a1c3e666e2ea860e1b1d5fc0

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ce6e33b3fa61a839ce25770b9245a5e163b9089e

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1a118a46f6883f108315b1b029eacc166ac5fced

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4d2dbd75e093751607861b566b1077db06bf590f

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=81759f0547e47277ab89cffd43c93df8914f450c

**Pull request merged**